### ISSUE TITLE
Lowered parameter with channels last dim order is actually stored as contiguous

### ISSUE THREAD

**Issue by MartinPavella** [pytorch/executorch#16464]
### 🐛 Describe the bug

I have played around with the channels last dim order (memory format) and discovered that trained parameters with this dim order are not actually stored as channels last in the `.pte` file. When I parse the `.pte` file (raw data), I can find the parameter's data stored in the contiguous dim order. Also, when I execute the model using `executor_runner` and debug the `kernels/portable/cpu/op_add.cpp` (which is the first operator to use the parameter in my model), I can see that the data is still contiguous, even though the parameter's `.dim_order()` returns `(0, 2, 3, 1)` (channels last dim order).

I have determined that the parameter data is contiguous by comparing it with the data from the PyTorch model before converting to channels last memory format and lowering to edge dialect. (`print()` statement in the code below)

I believe this is not the expected behavior and is probably the cause why my model is producing incorrect results.

Code to generate my model:
```python
import torch
from executorch.exir.capture._config import ExecutorchBackendConfig
from executorch.exir.program._program import to_edge_transform_and_lower
from torch.export import export


class AddModel(torch.nn.Module):

    def __init__(self):
        super().__init__()
        self.param = torch.nn.Parameter(torch.randn(1, 2, 3, 4))
        self.param.requires_grad = True

    def forward(self, x):
        return x + self.param


def to_executorch_program(model, example_input):
    exir_program_aten = torch.export.export(model, example_input, strict=True)

    partitioners = []
    edge_program_manager = to_edge_transform_and_lower(
        export(exir_program_aten.module(), example_input, strict=True),
        partitioner=partitioners,
    )

    return edge_program_manager.to_executorch(
        config=ExecutorchBackendConfig(extract_delegate_segments=False)
    )


if __name__ == "__main__":
    model = AddModel().eval()

    print("Contiguous flat data", model.param.data.flatten())
    model = model.to(memory_format=torch.channels_last)

    input_tensor = (torch.randn(1, 2, 3, 4).to(memory_format=torch.channels_last),)
    executorch_program_manager = to_executorch_program(model, input_tensor)

    executorch_program_manager.save("channels_last_dim_order_bug.pte")

```

The stored model is [attached](https://github.com/user-attachments/files/24449481/channels_last_dequantize_bug.zip).

### Versions

Collecting environment information...
PyTorch version: 2.10.0.dev20251025+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 11.4.0-2ubuntu1~20.04) 11.4.0
Clang version: Could not collect
CMake version: version 3.31.10
Libc version: glibc-2.31

Python version: 3.10.12 (main, Oct  1 2025, 11:13:47) [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-6.6.87.2-microsoft-standard-WSL2-x86_64-with-glibc2.31
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Byte Order:                           Little Endian
Address sizes:                        46 bits physical, 48 bits virtual
CPU(s):                               16
On-line CPU(s) list:                  0-15
Thread(s) per core:                   2
Core(s) per socket:                   8
Socket(s):                            1
NUMA node(s):                         1
Vendor ID:                            GenuineIntel
CPU family:                           6
Model:                                186
Model name:                           13th Gen Intel(R) Core(TM) i5-1350P
Stepping:                             2
CPU MHz:                              2188.810
BogoMIPS:                             4377.62
Virtualization:                       VT-x
Hypervisor vendor:                    Microsoft
Virtualization type:                  full
L1d cache:                            384 KiB
L1i cache:                            256 KiB
L2 cache:                             10 MiB
L3 cache:                             12 MiB
NUMA node0 CPU(s):                    0-15
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Mitigation; Clear Register File
Vulnerability Retbleed:               Mitigation; Enhanced IBRS
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI BHI_DIS_S
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology tsc_reliable nonstop_tsc cpuid tsc_known_freq pni pclmulqdq vmx ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb sha_ni xsaveopt xsavec xgetbv1 xsaves avx_vnni vnmi umip waitpkg gfni vaes vpclmulqdq rdpid movdiri movdir64b fsrm md_clear serialize flush_l1d arch_capabilities

Versions of relevant libraries:
[pip3] executorch==1.1.0a0+a069bba
[pip3] flake8==6.1.0
[pip3] flake8-breakpoint==1.1.0
[pip3] flake8-bugbear==24.4.26
[pip3] flake8-comprehensions==3.14.0
[pip3] flake8-plugin-utils==1.3.3
[pip3] flake8-pyi==23.5.0
[pip3] mypy==1.14.1
[pip3] mypy_extensions==1.1.0
[pip3] numpy==2.0.0
[pip3] optree==0.18.0
[pip3] pytorch_tokenizers==1.0.1
[pip3] torch==2.10.0.dev20251025+cpu
[pip3] torchao==0.14.0+git01849b2b1
[pip3] torchaudio==2.10.0.dev20251025+cpu
[pip3] torchdata==0.11.0
[pip3] torchsr==1.0.4
[pip3] torchtune==0.6.1
[pip3] torchvision==0.25.0.dev20251025+cpu
[conda] Could not collect

**Comment by JacobSzwejbka** [comment]
@Gasoonjia Does the dim order pass not actually permute the data aot?

**Comment by Gasoonjia** [comment]
it should be it might because of the serization.

https://github.com/pytorch/executorch/pull/16636 can you please retry on this PR? @MartinPavella

**Comment by lucylq** [comment]
Confirmed with https://github.com/pytorch/executorch/pull/16636 this case works, and stores tensors in channels-last format. Thanks @MartinPavella for raising the issue, will land that PR today.

**Comment by MartinPavella** [comment]
> it should be it might because of the serization.
> 
> [#16636](https://github.com/pytorch/executorch/pull/16636) can you please retry on this PR? [@MartinPavella](https://github.com/MartinPavella)

Thank you.
I can confirm that my model is working correctly with this PR.
